spec/hitnmiss/file_driver_spec.rb

```diff
@@ -0,0 +1,196 @@
+          subject.set('some_key', entity)
+          cache = JSON.parse(File.read(File.join(folder, "some_key")))
+          expect(cache['value']).to eq('some_value')
+          expect(cache.has_key?('expiration')).to eq(false)
+          expect(cache['fingerprint']).to eq('some-fingerprint')
+        end
+      end
+
+      context 'when given entity has a last_modified' do
+        it 'caches the given value with key, expiration, and last_modified' do
+          entity = Hitnmiss::Entity.new('some_value',
+                                        fingerprint: 'foofingerprint',
+                                        last_modified: '2016-04-14T11:00:00Z')
+          now = Time.now.utc
+          Timecop.freeze(now) do
+            subject.set('some_key', entity)
+          end
+          cache = JSON.parse(File.read(File.join(folder, "some_key")))
+          expect(cache['value']).to eq('some_value')
+          expect(cache['fingerprint']).to eq('foofingerprint')
+          expect(cache['last_modified']).to eq('2016-04-14T11:00:00Z')
+        end
+      end
+
+      it 'stores the updated_at timestamp in utc iso8601' do
+        entity = Hitnmiss::Entity.new('some_value')
+        now = Time.utc(2016, 4, 15, 13, 0, 0)
+        Timecop.freeze(now) do
+          subject.set('some_key', entity)
+        end
+        cache = JSON.parse(File.read(File.join(folder, "some_key")))
+        expect(cache['updated_at']).to eq('2016-04-15T13:00:00Z')
+      end
+    end
+  end
+
+  describe "#get" do
+    context "when key matches something cached" do
+      context "when we have passed the expiration" do
+        it "returns a Hitnmiss::Driver::Miss indicating a miss" do
+          cur_time = Time.now.utc
+          Timecop.freeze(cur_time) do
+            File.write(File.join(folder, "some_key"), { 'value' => 'foo', 'expiration' => (cur_time.to_i - 423423) }.to_json)
+            expect(subject.get('some_key')).to be_a(Hitnmiss::Driver::Miss)
+          end
+        end
+      end
+
+      context "when we have NOT passed the expiration" do
+        it "returns a Hitnmiss::Driver::Hit with the cached value" do
+          cur_time = Time.now.utc
+          Timecop.freeze(cur_time) do
+            File.write(File.join(folder, "some_key"), { 'value' => 'foo', 'expiration' => (cur_time.to_i + 234232) }.to_json)
+            hit = subject.get('some_key')
+            expect(hit).to be_a(Hitnmiss::Driver::Hit)
+            expect(hit.value).to eq('foo')
+          end
+        end
+      end
+    end
+
+    context "when key does not match something cached" do
+      it "returns a Hitnmiss::Driver::Miss indicating a miss" do
+        cache = { 'value' => 'foo', 'expiration' => 23 }
+        File.write(File.join(folder, "some_key"), cache.to_json)
+        expect(subject.get('some_other_key')).to be_a(Hitnmiss::Driver::Miss)
+      end
+    end
+  end
+
+  describe '#all' do
+    it 'returns only the values whose keys begin with the keyspace' do
```

Comment:
I think this also needs to take into consideration if the entities are expired or not, and only return values for cached entities that haven't already expired.